test/smc.jl

```diff
@@ -149,6 +150,34 @@
         @test all(isone(p.trajectory.f.x) for p in chains_pg)
         @test mean(x.logevidence for x in chains_pg) ≈ -2 * log(2) atol = 0.01
     end
+
+    @testset "Replay reference" begin
+        mutable struct Model <: AbstractMCMC.AbstractModel
+            a::Float64
+            b::Float64
+
+            Model() = new()
+        end
```

Comment:
IIRC one has to define structs at the top level.

Reply:
Seems like we define the structs locally in most of `test/smc.jl` but I can move it at the top module.
https://github.com/TuringLang/AdvancedPS.jl/blob/6979bbef862589a2dbe4cad062911409c733333e/test/smc.jl#L66-L98